### DISCUSSION TITLE
Install Fabric Mods

### DISCUSSION THREAD

**Discussion by YetiTheWinner** [Hexeption/MCP-Reborn#296] [Q&A]
Is there a way to simply install fabric mods in a mcp reborn client, so that they just run when the clients starts?

**Comment by APie357** [comment]
yesnt

You can _potentially_ install it, but it requires MultiMC/Prism Launcher. You could create an instance, load the jarmod, and then install fabric. You have to reobfuscate the client for the mappings of fabric to work.

And no, you can't do that in the dev environment.

**Reply by YetiTheWinner** [reply]
Is it possible to maybe decompile the mod and copy the folders in IntelliJ?

**Reply by APie357** [reply]
> Is it possible to maybe decompile the mod and copy the folders in IntelliJ?

You could probably get the source code from GitHub, but you would have to migrate the mappings to the official ones and inject the mixing yourself, not to mention the initial loading of the mod too.

**Reply by YetiTheWinner** [reply]
Okay. I wanted to create a launcheable client for a mod a friend of mine coded. First, I thought about coding a whole launcher, but it seemed to extensive, so I switched to coding a client. Is there any other way I can code a launcher or something similar with the mod included? I have full access to the mod.

**Reply by APie357** [reply]
You can create an instance in the official launcher without the fabric mods by following the section in the readme.

For your use case, I believe that this tool isn't for you and you should look into making a fabric mod. I found Kaupenjoe's modding tutorials to be very helpful, even though I don't use many of them and those tutorials don't cover client mods. If you would like to contact me, please add me on discord: `apie357`.

**Reply by YetiTheWinner** [reply]
Thanks a lot for your help, but I think you didn’t understand what I want to do. My friend and I coded a fabric mod and I wanted to code an own launcher just like Badlion or Lunar (obviously not as extensive) that includes the mod and maybe offers some options to add open-source performance mods. I‘m pretty sure that this tool is not what I‘m looking for, but I didn’t find an alternative and I don’t want to code it from scratch (I also don’t know how to). Do you know a method to do this?

**Reply by APie357** [reply]
This tool is definitely not for you. If you would like, I could help you with this, but I am currently unable to because my computer is broken and am still getting replacement parts. I would highly suggest that you add me on discord (`apie357`) and get it if you don't already have it. I would gladly help you.
